app/controllers/admin/merchants_controller.rb

```diff
@@ -0,0 +1,15 @@
+class Admin::MerchantsController < ApplicationController
+
+  def show
+    render file: "/public/404", status: 404 unless admin_user?
+    @current_user = User.find(params[:id])
+  end
+  
+  def update
```

Comment:
Line9 - Why is @current_user an instance variable?

Line10 - Why do we need to check `if admin_user?` for this line?  Could we instead add `render file: "/public/404", status: 404 unless admin_user?` as a `before_action` since no other users should have access to any of these controllers?

Line 12 - Can we include the user.name in the flash message?

Reply:
Added a before_action to all admin controllers (the method is stored in app_cont), 
added name to flash message